drivers: ieee802154: use DT_HAS* helpers and Kconfig simplification

gmarull opened the pull request:
- Make all drivers default to 'y' and dependent on being enabled in DT by using the DT_HAS* helpers.
- Remove redundant entries in Kconfig.defconfig files (not needed anymore)
- Make IEEE 802.15.4 drivers dependent on NETWORKING at the top level (simplifies already existing dependencies)

Partial work from https://github.com/zephyrproject-rtos/zephyr/pull/47938

gmarull committed:
drivers: ieee802154: depend on DT status and default to y

Make all drivers default to 'y' and dependent on being enabled in DT.
This will allow simplifying many samples/tests.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
kconfig: remove redundant IEEE 802.15.4 defaults or selections

All IEEE 802.15.4 drivers are now automatically enabled if they are
'okay' in DT and all of its dependencies are 'y', including
CONFIG_IEEE802154. This means individual driver enablement is not
necessary anymore in Kconfig.defconfig files or samples/tests. Boards
need to still make sure any dependencies are enabled, e.g. SPI bus in
some cases.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: ieee802154: IEEE802154 depends on NETWORKING

Previous to this change, each individual driver option depended on
NETWORKING. Instead, move dependency one level up.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

galak commented on drivers/ieee802154/Kconfig.uart_pipe:
```diff
@@ -3,7 +3,9 @@
 	bool "UART PIPE fake radio driver support for QEMU"
-	depends on (BOARD_QEMU_X86 || BOARD_QEMU_CORTEX_M3) && NETWORKING
+	default y
+	depends on (BOARD_QEMU_X86 || BOARD_QEMU_CORTEX_M3) && \
```
Do we know why this is limited to these qemu boards?  Could this be `QEMU_TARGET` instead?

gmarull replied:
@tbursztyka ^

Anyway, I think that is out of the scope of this PR, can be fixed in a follow-up PR if needed.

gmarull committed:
tests: drivers: build_all: ieee802154: remove mcr20a from spi.dtsi

This driver can't be compiled on all platforms as it's a NXP specific
driver (requires NXP HAL). Because spi.dtsi file is used by app.overlay
(and so enabled in multiple configs, including native_posix) it needs to
be removed. Note that MCR20A driver is already covered by the
ieee802154.build.mcr20a test case, which runs specifically on a NXP
board.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>
